Applying text alignment classes except when "undefined"

webmandesign opened the pull request:
## Description
Fixes #10558

## How has this been tested?
Please note that I don't have properly set up dev environment to create Gutenberg build so I couldn't really test this. Can someone please test the code? Thanks!

## Types of changes
Instead of checking for `center` align value for not applying the text alignment class, we check for `undefined`.

## Checklist:
- [ ] My code is tested.

webmandesign committed:
Applying text alignment classes except when "undefined"

Fixes issue #10558

ZebulanStanphill commented on packages/block-library/src/cover-image/index.js:
```diff
@@ -152,7 +152,7 @@ export const settings = {
 			const classes = classnames(
 				className,
-				contentAlign !== 'center' && `has-${ contentAlign }-content`,
+				contentAlign !== 'undefined' && `has-${ contentAlign }-content`,
```
I haven't tested the code, but this looks like it should probably be `undefined`, not `'undefined'`.

jorgefilipecosta replied:
I think we should use the real undefined instead of string here. I think the problem of receiving undefined strings happens because we have a default value of center on contentAlign attribute. I guess if we remove the default the problem gets solved.

jorgefilipecosta reviewed (commented):
Hi @webmandesign thank you for your contribution, I'm favorable to this change.
I think we should be careful here in order to avoid deprecations/migrations.
Previously all covers created had the default align of center but the attribute is not saved and no class was applied. This is equivalent to our new unset alignment state, so what I propose is that we remove the default alignment so existing blocks with center alignment are now automatically interpreted as covers without content align.
Doing this we have now a new state that previously was not possible the forced center alignment state.

webmandesign committed:
Using standard `undefined`, not a string

webmandesign committed:
Removing default value for `contentAlign`

webmandesign commented:
Hi @jorgefilipecosta ,

Thanks for your comments. I've went ahead and modified the code this way:
* Using real `undefined` instead of a string
* Removed default value setup for `contentAlign`

Please note that I don't have my dev environment set to build/test Gutenberg, so, could someone test these modifications to see if they work as expected? Thanks!

obenland reviewed (changes requested):
These classes were removed in #13822, I suggest closing this.

gziolo commented:
@jorgefilipecosta can you confirm that it can be closed?

jorgefilipecosta commented:
Hi @webmandesign I'm sorry for the delay I missed your answer in the middle of the notifications.
With the merge of nesting in cover block content alignment is managed by the InnerBlocks and the classes are not used anymore. So I'm closing this PR.
Thank you @webmandesign for this PR which otherwise would be highly demanded.
